pkg/xen-tools/patches-4.15.0/15-qemu-Set-the-affinity-of-QEMU-threads-according-to-t.patch

```diff
@@ -84,7 +84,7 @@ index b4c2359..44975eb 100644
 +
 +    /* Count the size of the necessary CPU_SET */
 +    max_pcpu = get_max_cpu_in_mask(cpumask);
-+    cpu_set_size = DIV_ROUND_UP(max_pcpu, BITS_PER_BYTE);
++    cpu_set_size = DIV_ROUND_UP(max_pcpu + 1, BITS_PER_BYTE);
```

Comment:
careful, __builtin_clz() (get_max_cpu_in_mask) is not defined for 0 and your function does not check the mask parameter. other than that looks good to me.

Reply:
It should not happen, as it's called only in the following case:
```
    if (cpu->cpumask)
        qemu_thread_set_affinity(cpu->thread, cpu->cpumask);
```

Reply:
Nice, but then at least explicitly assert in the function. Callers are not always careful, even this caller is only you :)

Reply:
Done